src/rust_api.rs

```diff
@@ -143,6 +152,45 @@ pub struct DnaSequence<T: NucleotideLike> {
     dna: Vec<T>,
 }
 
+#[cfg(feature = "serde")]
+impl<'de, T: NucleotideLike> serde::Deserialize<'de> for DnaSequence<T> {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        T: NucleotideLike,
+        D: serde::Deserializer<'de>,
+    {
+        struct Helper<T>(PhantomData<T>);
+        impl<'de, T: NucleotideLike> serde::de::Visitor<'de> for Helper<T> {
+            type Value = DnaSequence<T>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                write!(formatter, "a string")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                value
+                    .parse::<Self::Value>()
+                    .map_err(serde::de::Error::custom)
+            }
+        }
+
+        deserializer.deserialize_str(Helper(PhantomData))
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<T: NucleotideLike> serde::Serialize for DnaSequence<T> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.collect_str(self)
+    }
+}
+
```

Comment:
A sufficiently skilled (and bored) macro-writer could replace even this pair of impls with an invocation of `impl_stringlike!`, by making that macro smart about type parameters somehow. But that sounds a bit daunting to me, and the resulting macro would be even more daunting to read, so I will leave it "expanded".

I considered:
```rs
serde_utils::impl_stringlike!(DnaSequence<Nucleotide>);
serde_utils::impl_stringlike!(DnaSequence<NucleotideAmbiguous>);
```
but it would break code that is generic over `DnaSequence<T: Nucleotide>` and needs it to be (de)serializable.